optuna/storages/_journal/base.py

```diff
@@ -41,3 +42,36 @@ def append_logs(self, logs: List[Dict[str, Any]]) -> None:
         """
 
         raise NotImplementedError
+
+
+class BaseJournalLogSnapshot(abc.ABC):
+    """Optional base class for Journal storages.
+
+    Storage classes implementing this base class may work faster when
+    constructing the internal state from the large amount of logs.
+
+    Args:
+        snapshot_interval: An interval of trials and studies to save a snapshot.
+    """
+
+    def __init__(self, snapshot_interval: int) -> None:
+
+        self.snapshot_interval = snapshot_interval
```

Comment:
How about making it a class attribute and setting the default number instead of implementing a constructor method?

```suggestion
    snapshot_interval = 100
```

Reply:
Do you have any idea to document the class attribute?

Reply:
For class attributes, we can use `Attributes:` directive perhaps (See https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html for details).

Furthermore, on second thought, the following design might be clear to ensure that all child classes must set `snapshot_interval` argument.

```python
class BaseJournalSnapshot(abc.ABC):
    snapshot_interval: int

    @abc.abstractmethod
    def load_snapshot(...) -> bytes:
        ...

    @abc.abstractmethod
    def load_snapshot(...) -> bytes:
        ...

class JournalRedisStorage(BaseJournalStorage, BaseJournalSnapshot):
    def __init__(self, ..., snapshot_interval=100):
        self.snapshot_interval = snapshot_interval
```

@sile Do you have any opinion or suggestion about this?

Reply:
IMO, it is enough to document the `snapshot_interval` at the docstring of `BaseJournalSnapshot` using `Attributes:` directive, since the developers who want to implement a new journal storage backend must read the document of `BaseJournalStorage` and `BaseJournalSnapshot`.

Reply:
There seems to be several options for when to save the snapshot. To prevent this PR from being prolonged, it might be a good to make the `snapshot_interval` unchangeable in this PR.